### PR TITLE
Added Dovecot project

### PR DESCRIPTION
**Pull request by DavidKorczynski**
I added the Dovecot project with three initial fuzzers. Please follow the thread on Dovecots repo about progress: https://github.com/dovecot/core/pull/116 

By the looks of it they will provide an email soon that can get the bug reports, then I will update here. I made this pull request so we can follow progress both places.

### CODE REVIEW

File: projects/dovecot/build.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+make
+
+cd src/lib-imap
+
+echo $CFLAGS
+
+
+# fuzz-imap-utf7
+clang $CFLAGS -DHAVE_CONFIG_H -I. -I../..  -I../../src/lib -I../../src/lib-test -I../../src/lib-charset -I../../src/lib-mail   -std=gnu99 -fstack-protector-strong -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wall -W -Wmissing-prototypes -Wmissing-declarations -Wpointer-arith -Wchar-subscripts -Wformat=2 -Wbad-function-cast -Wno-duplicate-decl-specifier -Wstrict-aliasing=2   -MT test-imap-utf7.o -MD -MP -c -o fuzz-imap-utf7.o fuzz-imap-utf7.c
```
Comment (catenacyber) [inline projects/dovecot/build.sh]: The fuzz targets should be built by `make` cf https://github.com/dovecot/core/pull/117

This way, we do not have to modify oss-fuzz each time a new fuzz target is added to dovecot

Comment (DavidKorczynski) [reply projects/dovecot/build.sh]: Sounds good - I will update once @sirainen  finalises his thoughts on how it will be added in Dovecot!

**Comment by catenacyber** [comment]
Looks like build.sh should begin with `cd core`

**Comment by DavidKorczynski** [comment]
Hey @mbarbella-chromium ! Could you please review again? In the end the maintainers of Dovecot were very enthusiastic and did a great job integrating it nicely into the Dovecot building process (https://github.com/dovecot/core/pull/116 and https://github.com/dovecot/core/pull/117/files). This should be a good setup for future fuzzing of Dovecot too!

**Comment by JulianVolodia** [comment]
Hey! I found dovecot was not #402 marked as solved - @DavidKorczynski maybe worth saying it there :)))

**Comment by Dor1s** [comment]
> Hey! I found dovecot was not #402 marked as solved - @DavidKorczynski maybe worth saying it there :)))

yay, marked as Done! Thanks a lot!

**Pull request merged**